lib/rules/require-spaces-in-for-statement.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * Requires spaces inbetween for statement.
+ *
+ * Type: `Boolean`
+ *
+ * Values: `true` to requires spaces inbetween for statement.
+ *
+ * #### Example
+ *
+ * ```js
+ * "requireSpacesInForStatement": true
+ * ```
+ *
+ * ##### Valid
+ *
+ * ```js
+ * for(var i = 0; i<l; i++) {
+ *     x++;
+ * }
+ * ```
+ *
+ * ##### Invalid
+ *
+ * ```js
+ * for(var i = 0;i<l;i++) {
+ *     x++;
+ * }
+ * ```
+ *
+ * ```js
+ * for(var i = 0; i<l;i++) {
+ *     x++;
+ * }
+ * ```
+ *
+ * ```js
+ * for(var i = 0;i<l; i++) {
+ *     x++;
+ * }
+ * ```
+ */
+
+var assert = require('assert');
+
+module.exports = function() {};
+
+module.exports.prototype = {
+    configure: function(requireSpacesInForStatement) {
+        assert(
+            requireSpacesInForStatement === true,
+            'requireSpacesInForStatement option requires true value or should be removed'
+        );
+    },
+
+    getOptionName: function() {
+        return 'requireSpacesInForStatement';
+    },
+
+    check: function(file, errors) {
+        file.iterateNodesByType('ForStatement', function(node) {
+            if (node.test) {
+                var testToken = file.getTokenByRangeStart(node.test.range[0]);
+                var prevToken1 = file.getPrevToken(testToken);
+                errors.assert.whitespaceBetween({
+                    token: prevToken1,
+                    nextToken: testToken,
+                    spaces: 1,
+                    message: 'Missing space after semicolon'
+                });
+            }
+            if (node.update) {
+                var updateToken = file.getTokenByRangeStart(node.update.range[0]);
+                var prevToken2 = file.getPrevToken(updateToken);
```

Comment:
`prevToken2 ` - not a fun of this kind of naming, maybe something more meaningful would be better?